Whiskey/Tasks/Publish-WhiskeyBuildMasterPackage.ps1

```diff
@@ -15,19 +15,48 @@ function Publish-WhiskeyBuildMasterPackage
 
     Set-StrictMode -Version 'Latest'
     Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
-
+    
     $applicationName = $TaskParameter['ApplicationName']
     if( -not $applicationName )
     {
         Stop-WhiskeyTask -TaskContext $TaskContext -Message ('Property ''ApplicationName'' is mandatory. It must be set to the name of the application in BuildMaster where the package should be published.')
     }
-
-    $releaseName = $TaskParameter['ReleaseName']
+    
+    $branch = $TaskContext.BuildMetadata.ScmBranch
+    $releaseName = $null
+    $startAtStage = $null
+    $skipDeploy = $null
+    
+    if( $TaskParameter['DeployTo'] )
+    {
+        $idx = 0
+        Write-Verbose -Message ('DeployTo')
+        :deploy foreach( $item in $TaskParameter['DeployTo'] )
+        {
+            foreach( $wildcard in $item['BranchName'] )
+            {
+                if( $branch -like $wildcard )
+                {
+                    Write-Verbose -Message ('               {0}     -like  {1}' -f $branch,$wildcard)
+                    $releaseName = $item['ReleaseName']
+                    $startAtStage = $item['StartAtStage']
+                    $skipDeploy = $item['SkipDeploy']
+                    break deploy

```

Comment:
Can't you use `continue`?

Reply:
I don't think it will work in this case because we need to exit out of the outer loop once we've found our first match. In the example `DeployTo` property configuration below, if we are on branch `JeffTasks`:

- `break deploy` will map to a `ReleaseName` of `gitflow`
- `continue` will map to a `ReleaseName` of `notGitflow`

DeployTo:
- BranchName: JeffTasks
  ReleaseName: gitflow
- BranchName:
      - develop
      - Jef*
  ReleaseName: notGitflow